src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,121 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold = Math.round(DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> processingNode = table[index];
+        if (processingNode == null) {
+            table[index] = new Node<K, V>(key, value, null);
+            size++;
+        } else {
+            while (processingNode.next != null) {
+                if (areValuesEqual(processingNode.key, key)) {
+                    processingNode.value = value;
+                    return;
+                }
+                processingNode = processingNode.next;
+            }
+            if (areValuesEqual(processingNode.key, key)) {
+                processingNode.value = value;
+                return;
+            }
+            processingNode.next = new Node(key, value, null);
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = getIndex(key);
+        Node<K, V> processingNode = table[index];
+        while (true) {
+            if (processingNode == null) {
+                return null;
+            } else if (areValuesEqual(processingNode.key, key)) {
+                return processingNode.getValue();
+            } else {
+                processingNode = processingNode.next;
+            }
+        }
     }
 
     @Override
     public int getSize() {
+        return size;
+    }
+
+    private boolean areValuesEqual(K key, K current) {
+        return (current != null && current.equals(key))
+                || (current == null && key == null);
+    }
+
+    private void resize() {
+        size = 0;
+        int newCapacity = table.length * 2;
+        Node<K, V>[] oldTable = table;
+        table = new Node[newCapacity];
+        threshold = Math.round(newCapacity * DEFAULT_LOAD_FACTOR);
+        for (Node<K, V> kvNode : oldTable) {
+            while (kvNode != null) {
+                this.put(kvNode.key, kvNode.value);
+                kvNode = kvNode.next;
+            }
+        }
+    }
+
+    @Override
+    public String toString() {
```

Comment:
This is the wrong implementation, nested nodes will not be displayed, I suggest just to remove this method